### DISCUSSION TITLE
The Backrooms mod Port discussion

### DISCUSSION THREAD

**Discussion by ghost** [cataclysmbn/Cataclysm-BN#5112] [General]
https://github.com/CleverRaven/Cataclysm-DDA/pull/60121

Looking over the mod in current DDA as well as the original pull I see no EOC content to worry about converting to lua, but I mainly worry our mapgen isn't up to date to handle this. Specifically we're missing the flag ERASE_ALL_BEFORE_PLACING_TERRAIN it seems their mapgen is mostly similar to ours however. I think we can safely try to pull the whole thing into BN, remove relevant DDA fields such as length or pocket data, convert teleportation to our magic system in case theirs is different, and see what happens with the mapgen.

Any relevant DDA PR's to mapgen would be appreciated if someone knows them offhand.

**Comment by scarf005** [comment]
related DDA PRs regarding `ERASE_ALL_BEFORE_PLACING_TERRAIN`: https://github.com/CleverRaven/Cataclysm-DDA/commits/bf02871518140a0da7af16ce80b93c3adec6cf96/src/jmapgen_flags.h

**Comment by scarf005** [comment]
also seems like there's already an BN mod by [original author](https://old.reddit.com/r/cataclysmdda/comments/p4m20y/i_made_an_infinite_backrooms_mod/): https://github.com/TGWeaver/Backrooms-CBN-mod

**Comment by ghost** [comment]
Didn't realize, that should probably go in mod links then.

**Comment by Maxwellneo** [comment]
Welp... It needs sort of a port, since it really lacks all needed stuff to run on modern BN

**Comment by scarf005** [comment]
closing as https://github.com/cataclysmbn/Cataclysm-BN/pull/8500 opened
